### PR TITLE
Enabling Helix perf test runs for CoreCLR

### PR DESCRIPTION
**Pull request by dsgouda**
- Enabling discovery and archival of individual perf tests and uploading them as workitems
- Ensuring CoreCLR product is packaged and uploaded as correlation payload for the testing

@rahku @brianrob  @jhendrixMSFT @MattGal

### CODE REVIEW

**Comment by dsgouda** [comment]
@dotnet-bot Test this please.

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

File: tests/runtest.cmd [hunk 1]
```diff
@@ -191,6 +192,19 @@ if errorlevel 1 (
     exit /b 1
 )
```
Comment (rahku) [inline tests/runtest.cmd]: It seems like perf tests will also run as part of functional test runs. I don't think we want that

File: tests/src/dir.targets [hunk 2]
```diff
@@ -75,6 +75,11 @@
     <MSBuild Projects="@(ProjectReference)" />
     <MakeDir Condition="'$(CLRTestKind)' == 'RunOnly'" ContinueOnError="false" Directories="$(OutputPath)" />
   </Target>
+ 
+  <Target Name="CopyJsonProjectFiles" AfterTargets="Build">  
+    <!-- Post build copy project json files so we can generate assembly lists for the projects -->  
+    <Copy SourceFiles="$(ProjectJson);$(ProjectLockJson)" DestinationFolder="$(OutputPath)" />
```
Comment (rahku) [inline tests/src/dir.targets]: why do you need to copy both project,json & project.lock.json. I believe only lock.json would be sufficient

File: tests/helixperftasks.targets [hunk 3]
```diff
@@ -0,0 +1,89 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="GetPerfTestAssemblies" AssemblyFile="$(ToolsDir)net45/Microsoft.DotNet.Build.CloudTestTasks.dll"/>
+  <Import Project="helix.targets" />
```
Comment (rahku) [inline tests/helixperftasks.targets]: is this required?

File: tests/runtest.proj [hunk 4]
```diff
@@ -412,6 +412,21 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
...
 
+  <Target Name="RunPerfTests" Condition="'$(Performance)'=='true'">  
+    <Message Text="Executing steps for perf tests" Importance="High"/>  
+    
+    <!-- generate project.json for runtime dependency -->  
+    <MSBuild Projects="$(MSBuildProjectFile)"  
+             Targets="CreateTestRuntimeJsonFile"/>  
+     
+    <!-- Package the product -->  
+    <MSBuild Projects="$(MSBuildProjectFile)" Targets="CreateTestOverlay" />
```
Comment (rahku) [inline tests/runtest.proj]: I didn't get why is this required?

File: tests/src/dir.targets (hunk 2 above)
Comment (dsgouda) [inline tests/src/dir.targets]: I realize its an overkill but it helps debugging on the test machine

File: tests/runtest.proj (hunk 4 above)
Comment (dsgouda) [inline tests/runtest.proj]: This is to copy the product to the payload directory so that we can package it as a correlation payload

File: tests/runtest.cmd (hunk 1 above)
Comment (AndyAyersMS) [inline tests/runtest.cmd]: We at least want the jit perf tests to also be run as normal correctness tests.

File: tests/helixperftasks.targets (hunk 3 above)
Comment (dsgouda) [inline tests/helixperftasks.targets]: This is so that we can refer to the "AddFilesToAssemblyList" target

File: tests/runtest.cmd (hunk 1 above)
Comment (rahku) [inline tests/runtest.cmd]: Yes that was not what I meant. I did not mean to exclude perf tests from normal test runs. I was referring to the fact the script does not exit when someone does runtest.cmd (to only do functional test runs locally). It goes on to also send perf runs to helix.

File: tests/runtest.proj (hunk 4 above)
Comment (rahku) [inline tests/runtest.proj]: Product actually should be coming from packages folder and not from this task. I think you can remove this. creation of correlation payload would work fine without this task. It happens in cloudtest.targets.

File: tests/src/dir.targets (hunk 2 above)
Comment (rahku) [inline tests/src/dir.targets]: I would be OK if you did this only for perf tests but since this happens for all the tests. This would add up.

File: tests/helixperftasks.targets (hunk 3 above)
Comment (rahku) [inline tests/helixperftasks.targets]: That target would go away soon. It was added because of xunitwrappers required for functional runs. Since for perf runs you do not have that problem....it should not be required ideally.

Comment (rahku) [inline tests/helixperftasks.targets]: It would be better to not take a dependency on that.

File: tests/runtest.proj (hunk 4 above)
Comment (dsgouda) [inline tests/runtest.proj]: If I understand correctly the runtime dependencies get copied over from here
https://github.com/dotnet/coreclr/blob/43cc989069c400c939f1d4df878876523bf336ed/tests/publishdependency.targets#L15
If I don't call this target the assembly list generated has empty "corerun" element

Comment (rahku) [inline tests/runtest.proj]: corerun.exe dependency is add in src\common\test_runtime\project.json
"Microsoft.NETCore.TestHost": "1.0.0-rc2-23816", 
Above package should bring in corerun.exe

File: tests/runtest.cmd [hunk 5]
```diff
@@ -191,6 +192,23 @@ if errorlevel 1 (
...
 
+if not defined %__PerfTests% (
+	exit /b 1
+)
```
Comment (rahku) [inline tests/runtest.cmd]: now with your this change on functional runs I will not get the success message that gets printed when all tests pass successfully. Look at messages at line 219.

**Comment by rahku** [comment]
Looks good except for above comment.

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot Test this please

**Comment by dsgouda** [comment]
@dotnet-bot  test Windows_NT x64 Release Priority 1 Build and Test
